examples/experimental/dagster-airlift/dagster_airlift/core/serialization/serialized_data.py

```diff
@@ -103,7 +103,8 @@ def all_mapped_tasks(self) -> Dict[AssetKey, List[MappedAirflowTaskData]]:
         return {item.asset_key: item.mapped_tasks for item in self.key_scoped_data_items}
 
     def task_ids_in_dag(self, dag_id: str) -> AbstractSet[str]:
-        return set(self.dag_datas[dag_id].task_handle_data.keys())
+        """This returns all task ids in a the dag, whether or not there exists any mappings."""
+        return set(self.dag_datas[dag_id].task_infos.keys())
```

Comment:
Not sure I understand why this is necessary, or how it applies to the automapping functionality.

In my mind, one of the main reasons someone would want to turn off task automapping is perf concerns. It feels bad that we're retrieving the same amount of data regardless of how much we actually need.

Once we have the task mappings associated with the constructed specs, it shouldn't be hard to get the exact set of tasks that have a spec associate with them right. Which is exactly the number of task instances that we want to query.

Reply:
It applies to the automapping functionality because this map is built in the `fetch_from_state` phase of the process. It is built from the explicit_defs passed in by the user. 

In this PR, the new automapped _definitions_ (along with their task mappings) are only created in the `defs_from_state`portion of the process, after the serialized data has been created. 

The way to make it so this change is necessary is to change create the automapped definitions, merge them with the explicit defs, and then put them through the machinery to build the serialized data.

Reply:
This code was completely eliminating so resolving this thread